source/common/stats/stats_impl.h

```diff
@@ -333,40 +338,77 @@ class GaugeImpl : public Gauge, public MetricImpl {
   // Stats::Gauge
   virtual void add(uint64_t amount) override {
     data_.value_ += amount;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void dec() override { sub(1); }
   virtual void inc() override { add(1); }
   virtual void set(uint64_t value) override {
     data_.value_ = value;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void sub(uint64_t amount) override {
     ASSERT(data_.value_ >= amount);
     ASSERT(used());
     data_.value_ -= amount;
   }
   virtual uint64_t value() const override { return data_.value_; }
-  bool used() const override { return data_.flags_ & RawStatData::Flags::Used; }
+  bool used() const override { return data_.flags_ & Flags::Used; }
 
 private:
   RawStatData& data_;
   RawStatDataAllocator& alloc_;
 };
 
+/**
+ * Implementation of HistogramStatistics for circllhist.
+ */
+class HistogramStatisticsImpl : public HistogramStatistics {
+public:
+  HistogramStatisticsImpl() {}
+
+  HistogramStatisticsImpl(histogram_t* histogram_ptr) {
+    hist_approx_quantile(histogram_ptr, const_cast<double*>(quantiles_in_),
+                         ARRAY_SIZE(quantiles_in_), quantiles_out_);
+  }
+
+  std::string summary() const override {
+    return fmt::format("P0: {} , P25: {}, P50: {}, P75: {}, P90: {}, P95: {}, P99: {}, P100: {}",
+                       quantiles_out_[0], quantiles_out_[1], quantiles_out_[2], quantiles_out_[3],
+                       quantiles_out_[4], quantiles_out_[5], quantiles_out_[6], quantiles_out_[7],
+                       quantiles_out_[8]);
+  }
+};
+
 /**
  * Histogram implementation for the heap.
  */
 class HistogramImpl : public Histogram, public MetricImpl {
 public:
   HistogramImpl(const std::string& name, Store& parent, std::string&& tag_extracted_name,
                 std::vector<Tag>&& tags)
-      : MetricImpl(name, std::move(tag_extracted_name), std::move(tags)), parent_(parent) {}
+      : MetricImpl(name, std::move(tag_extracted_name), std::move(tags)), parent_(parent) {
+    cumulative_statistics_.reset(new HistogramStatisticsImpl());
```

Comment:
init in initializer list. Do these need to be pointers at all? Can't we just allocate them directly as members?